### PR TITLE
adding error message when configure_opentelemetry fails

### PR DESCRIPTION
**Pull request by codeboten**

### CODE REVIEW

File: src/opentelemetry/launcher/configuration.py [hunk 1]
```diff
@@ -289,7 +289,13 @@ def _validate_service_name(service_name: Optional[str]):
 
 class LightstepLauncherInstrumentor(BaseInstrumentor):
     def _instrument(self, **kwargs):
-        configure_opentelemetry()
+        try:
+            configure_opentelemetry()
+        except InvalidConfigurationError as err:
+            _logger.error(
```
Comment (ocelotl) [inline src/opentelemetry/launcher/configuration.py:295]: Would it be better to use [`exception`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) instead?

File: src/opentelemetry/launcher/configuration.py [hunk 2]
```diff
@@ -94,6 +94,7 @@ def configure_opentelemetry(
     log_level: str = _OTEL_LOG_LEVEL,
     span_exporter_insecure: bool = _OTEL_EXPORTER_OTLP_SPAN_INSECURE,
     metric_exporter_insecure: bool = (_OTEL_EXPORTER_OTLP_METRIC_INSECURE),
+    auto_instrumented: bool = False,
```
Comment (kayousterhout) [inline src/opentelemetry/launcher/configuration.py:97]: does Python have a thing where we can put an _ in front of this to make it private?

Comment (codeboten) [reply src/opentelemetry/launcher/configuration.py]: it does, I didn't realize this would work for method arguments, good suggestion!

Comment (ocelotl) [reply src/opentelemetry/launcher/configuration.py]: I also think using private arguments is the best way to go here. Just for the record, we can use function attributes if we don't want to add another argument, but for this situation, it is not so "clean" as the variables that affect the call are not in the call themselves:

```python
def configure_opentelemetry():
    if hasattr(configure_opentelemetry, "_auto_instrumented"):
        print("auto instrumented")
    else:
        print("not auto instrumented")

configure_opentelemetry()

configure_opentelemetry._auto_instrumented = True

configure_opentelemetry()
```
```
not auto instrumented
auto instrumented
```

Comment (kayousterhout) [reply src/opentelemetry/launcher/configuration.py]: oo I learned a lot here - did not know about those function attributes!!  Glad it worked to make this private

**Pull request merged**